### PR TITLE
chore: update npm_translate_lock docs to include info about compatabilty with pnpm outside of Bazel

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

File: npm/npm_import.bzl [hunk 1]
```diff
@@ -79,6 +79,33 @@ def npm_translate_lock(
...
+    If you hit import/require runtime failures in 3rd party npm packages when using rules_js that you do not get
+    outside of Bazel when using pnpm, we recommend adding `hoist=false` to your `.npmrc`:
+
+    ```
+    echo "hoist=false" >> .npmrc
+    ```
+
+    This will prevent pnpm from creating the hidden `node_modules/.pnpm/node_modules` folder with hoisted
+    dependencies which allow for packages to depend on "phantom" undeclared dependencies. In most cases, the failure
+    you see in rules_js will be reproducable outside of Bazel when hoisting is disabled.
```
Comment (alexeagle) [inline npm/npm_import.bzl:96]: nit: check spelling "reproducible"

File: npm/npm_import.bzl [hunk 2]
```diff
@@ -79,6 +79,33 @@ def npm_translate_lock(
     Thus it is performant to convert a very large pnpm-lock.yaml file without concern for
     users needing to fetch many unnecessary packages.
 
+    **Compatabilty with pnpm***
+
+    The node_modules layed out by rules_js should be bug-for-bug compatible with the node_modules that
```
Comment (alexeagle) [inline npm/npm_import.bzl:84]: nit: layed -> laid?

File: npm/npm_import.bzl [hunk 3]
```diff
@@ -79,6 +79,33 @@ def npm_translate_lock(
     Thus it is performant to convert a very large pnpm-lock.yaml file without concern for
     users needing to fetch many unnecessary packages.
 
+    **Compatabilty with pnpm***
```
Comment (alexeagle) [inline npm/npm_import.bzl:82]: should add some note to the faq or migration.md so ppl know to look here

**Review by alexeagle — APPROVED** [review]

File: npm/npm_import.bzl (hunk 1 above)
Comment (gregmagolan) [reply npm/npm_import.bzl]: hmmm. my spellcheck plugin in VSCode seems to be malfunctioning

File: npm/npm_import.bzl (hunk 3 above)
Comment (gregmagolan) [reply npm/npm_import.bzl:82]: ✅

**Pull request merged**